src/serializers/decoders/current/v11/decode_graph_schema.c

```diff
@@ -48,18 +50,25 @@ static Schema *_RdbLoadSchema(RedisModuleIO *rdb, SchemaType type) {
 		uint fields_count = RedisModule_LoadUnsigned(rdb);
 		for(uint i = 0; i < fields_count; i++) {
 			char *field = RedisModule_LoadStringBuffer(rdb, NULL);
-			Schema_AddIndex(&idx, s, field, index_type);
+			if(!already_loaded)
+				Schema_AddIndex(&idx, s, field, index_type);
 			RedisModule_Free(field);
 		}
 
-		if(index_type == IDX_FULLTEXT) {
+		if(index_type == IDX_FULLTEXT && !already_loaded) {
 			idx->language = language;
 			idx->stopwords = stopwords;
 		}
 
 		index_count -= fields_count;
```

Comment:
Need to make a few changes to the way indexes are formatted, I think `index_count` should reflect the number of indexes and `fields_count` should reflect the number of fields within a given index.
`index_count -= fields_count;` is confusing.